lib/facter/currentuser.rb

```diff
@@ -0,0 +1,60 @@
+require 'facter'
+
+## Define our Facts as nil first
+username = nil
+name = nil
+domain = nil
+sid = nil
+
+## Fact for full username of the currently/recently logged in user... usually helpful for Endpoints
+Facter.add("windows_currentuser_username") do
+  confine :kernel => 'windows'
+  setcode do
+    require 'win32ole'
+    wmi = WIN32OLE.connect('winmgmts://./root/CIMV2')
+    ## First find out
+    query1 = "select Username from win32_computersystem"
+    wmi.ExecQuery(query1).each do |data|
+      if data.Username.nil?
+        username = nil
+      else
+        domain, name = data.Username.split("\\")
+        username = data.Username
+      end
+
+    end
+    if username.nil?
+      sid = nil
+    else
+      query2 = "select sid from win32_useraccount where name='"+ name +"' and domain='"+ domain +"'"
+      wmi.ExecQuery(query2).each do |data|
+        sid = data.SID
+      end
+    end
+    username

```

Comment:
@joshcooper thanks for the hint on `id`, however it will pull in the id of the process that is running puppet, not the current console user (which is handy for "HKCU" support).

I'l look at using the sid converter instead of wmi, but may still rely on win32_computersystem to supply the "username" of the current console user.

Also i'll add support for 2 parameters `user` and `domain`, and use the sid conversion instead of WMI.  should simplify things for HKU use cases.

Thanks for the advice bud.